cmd/otelcol/Dockerfile

```diff
@@ -1,18 +1,15 @@
-ARG SMART_AGENT_RELEASE=v5.9.1
```

Comment:
Don't we need the default value for `SMART_AGENT_RELEASE`?

Reply:
It's set in the Makefile so removed it from here so it's not defined in multiple places.